### PR TITLE
Major update for all packages and separate packages for each chapter

### LINKED ISSUE #4: Add support for Git LFS
It's quite commonly used for mobile projects, so it would be good to have it by default.

```
brew install git-lfs
```

### LINKED ISSUE #10: Add support for AWS CLI
This is more a nice-to-have but it's such a pain to install properly that it would be a good default.

The recommended way (at least how I made it work) is to [the bundled installer](https://docs.aws.amazon.com/cli/latest/userguide/cli-install-macos.html#install-bundle-macos). Using `pip` failed several times (and their doc is out-dated as we should use `pip3`).

Since `laptop` already includes the Heroku CLI, this would come as a nice addition.

### LINKED ISSUE #11: Create default postgres user
Currently, when installing PostgreSQL service with brew, it creates a user matching the system user. Instead, it should create the generic user `postgres`:

```
/usr/local/opt/postgres/bin/createuser -s postgres
```

### LINKED ISSUE #16: Add Github CLI
## Why

Since we recently migrated to Github, https://cli.github.com/ would be a great addition to our toolset.
 
## Who Benefits?

All the developers.

### LINKED ISSUE #18: Add iOS developer specific apps
## Needed

- Xcode
- RVM
- cocoapods gem
- figma
- homebrew
- Proxyman
- Atom or some text editor
- iTerm2
- Sourcetree
- go2shell
- Spotify
- chrome
- slack
- Keybase
- 1password

 
## Who Benefits?

iOS chapter

### LINKED ISSUE #19: Add Web developer specific apps
## Needed

- postgres
- redis
- Heroku
- AWS CLI
- Github CLI
- imagemagick
- docker
- yarn
- nvm
 
## Who Benefits?

Web chapter

### LINKED ISSUE #20: Add general apps
## Needed

- git
- Keybase
- 1Password
- Slack
- Google Chrome
- zsh (with oh my zsh)
- gpg2 (for installing RVM)
- github cli
- gpg-suite
- rvm
- nvm
- bundler
- postman
- vscode
- iTerm2
- Skitch
 
## Who Benefits?

All chapters

### PR DESCRIPTION
**Pull request by hoangmirs**
## What happened

- Split the packages into multiple groups
- Add questions before installing packages
- This closes #4, closes #10, closes #11, closes #16, closes #18, closes #19 and closes #20 
 
## Insight

- Move the packages that are used by chapters to different functions. Merge all of them into Brewfile and install them together by running `brew bundle`
- Some general apps could be installed before running this script, so added some guards to skip them
- Add some questions to collect what to be installed at the beginning of the script
- You can check README to know what will be added/removed 
- Create default postgres user
 
## Proof Of Work

Not sure, I got an error when running `source ~/.rvm/scripts/rvm`

### CODE REVIEW

**Review by olivierobert — APPROVED** [review]
Looks good 🤩

File: README.md [hunk 1]
```diff
@@ -26,50 +26,109 @@ versions are welcome.
...
+* [Github CLI] GitHub’s official command line tool
 * [Homebrew] for managing operating system libraries
-* [ImageMagick] for cropping and resizing images
-* [Postgres] for storing relational data
 * [RVM] for managing versions of Ruby
 * [NVM] for managing versions of Node.JS
-* [Redis] for storing key-value data
-* [Ruby] stable for writing general-purpose code
-* [Tmux] for saving project state and switching between projects
+* [The Silver Searcher] a code-searching tool similar to ack, but faster
```
Comment (suho) [inline README.md]: This is a default tool, but I think I'm not using this, can we remove it from default setup

File: README.md [hunk 2]
```diff
@@ -26,50 +26,109 @@ versions are welcome.
...
 [The Silver Searcher]: https://github.com/ggreer/the_silver_searcher
-[Tmux]: http://tmux.sourceforge.net/
 [Zsh]: http://www.zsh.org/
 [oh-my-zsh]: http://ohmyz.sh/
-[Docker]: https://www.docker.com/
-[Google Chrome]: https://www.google.com/chrome/
-[Slack]: https://www.slack.com/
-[Keybase]: https://keybase.io/
+[VS Code]: https://code.visualstudio.com/
+[Watchman]: https://github.com/facebook/watchman
```
Comment (suho) [inline README.md]: I think for iOS developer, we don't need this tool, it should be removed from the default list

**Review by suho — COMMENTED** [review]
@hoangmirs Please check my comments, thank you

File: mac [hunk 3]
```diff
@@ -89,132 +117,192 @@ gem_install_or_update() {
...
+    cask "1password-cli"
+    cask "skitch"
+    cask "postman"
+    cask "iterm2"
+
+    # Unix
+    brew "universal-ctags", args: ["HEAD"]
+    brew "git"
+    brew "openssl"
+    brew "the_silver_searcher"
```
Comment (hoangmirs) [inline mac]: Do we need `the_silver_searcher/watchman`? @olivierobert 
Can we remove them?
CC @suho

File: mac [hunk 4]
```diff
@@ -89,132 +117,192 @@ gem_install_or_update() {
...
-  fancy_echo "Installing Google Chrome ..."
-  brew cask install google-chrome
-  fancy_echo "Google Chrome installed."
-fi
+  tee -a "$HOME/.Brewfile" <<-EOF
+    # iOS
+    brew "cocoapods"
+    cask "figma"
+    cask "proxyman"
+    cask "sourcetree"
```
Comment (byhbt) [inline mac:248]: Do we need sourcetree as default? 🤔

Comment (hoangmirs) [reply mac:248]: @byhbt Do you need it?
It was requested by iOS chapter, so I put it only on iOS group.
I think in Web chapter, we almost use git CLI, so I think it's redundant for us

**Comment by byhbt** [comment]
hi @hoangmirs @olivierobert !
I would recommend to install some default aliases, it's pretty handy. What do you think?
For example: https://laravel-news.com/bash-aliases

File: mac (hunk 4 above)
Comment (caherrerapa) [reply mac:248]: As we moved to github probabl makes more sense to install github desktop. I had to use it when I did normal web not ruby for example

File: README.md [hunk 5]
```diff
@@ -26,50 +26,109 @@ versions are welcome.
...
+[gpg2]: https://gnupg.org/
+[libyaml]: https://github.com/yaml/libyaml
+[coreutils]: https://www.gnu.org/software/coreutils
+
+### Web
+* [Ruby] stable for writing general-purpose code
```
Comment (caherrerapa) [inline README.md]: Ruby is also used by iOS afaik (e.g. cocoapods)

File: README.md [hunk 6]
```diff
@@ -82,17 +141,16 @@ For example:
 ```sh
 #!/bin/sh
 
+cask "dropbox"
```
Comment (caherrerapa) [inline README.md:146]: I think we don't use dropbox for file sharing

Comment (hoangmirs) [reply README.md:146]: Yes, this line is an example for customizations.

File: mac (hunk 4 above)
Comment (hoangmirs) [reply mac:248]: I added Github Desktop to general group

File: README.md (hunk 5 above)
Comment (hoangmirs) [reply README.md]: Yes, I moved the description to general group. Ruby is installed by default, along with RVM

File: mac (hunk 3 above)
Comment (olivierobert) [reply mac]: It can be removed 😃

**Comment by olivierobert** [comment]
@hoangmirs Please update the descriptions with mention: `closes #issue IDs` so that it closed all open issues

**Review by byhbt — APPROVED** [review]

**Comment by olivierobert** [comment]
@suho Please have a second round of reviews ✌️

**Review by suho — APPROVED** [review]
LGTM

**Comment by olivierobert** [comment]
@hoangmirs Please tag this as the 2.0.0 version. The previous version could be tagged 1.0.0 retroactively based on the previous commits history.

**Pull request merged**